www/components/MyST.tsx

```diff
@@ -0,0 +1,270 @@
+import * as md from "mdast";
+import { MyST } from "mystjs";
+import React, { ReactElement, useContext } from "react";
+import { Literal, Node } from "unist";
+
+export interface MySTMarkupProps extends MySTRoles {
+  source: string;
+}
+
+export interface MySTRoles {
+  roles?: (role: MySTRole) => ReactElement;
+}
+
+const MySTContext = React.createContext<MySTRoles>({ roles: (_) => <></> });
+
+export default function MySTMarkup({ source, roles }: MySTMarkupProps) {
+  const parser = new MyST();
+  const tree = parser.parse(source);
+  return (
+    <MySTContext.Provider value={{ roles }}>
+      <MySTParent node={tree} />
+    </MySTContext.Provider>
+  );
+}
+
+interface DispatchProps<T extends Node> {
+  key: number;
+  node: T;
+}
+
+export interface MySTRole extends Literal {
+  type: "mystRole";
+  name: string;
```

Comment:
This is where I'd usually recommend making this a kinda-enum with something like `name: "role-1" | "role-2" | ...`, except: is that sensible? Do we actually want to restrict / explicitly define the list of supported roles like this? Even better would be parsing this into some inflated version (ADT FTW) explicitly describing WTH is going on in a given `MySTRole`.

But also: first make it work, then make it scale.